File: 卡码/代码随想录/Java/src/com/kurumi/codeThoughts/二叉树/_04_二叉搜索树属性/LeetCode_0538.java
package com.kurumi.codeThoughts.二叉树._04_二叉搜索树属性;

import com.kurumi.codeThoughts.entity.TreeNode;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 538. 把二叉搜索树转换为累加树
 */
public class LeetCode_0538 {
    static class Solution {
        int pre = 0;

        public void traversal(TreeNode root) {
            if (root == null)
                return;
            traversal(root.right);
            root.val += pre;
            pre = root.val;
            traversal(root.left);
        }

        public TreeNode convertBST(TreeNode root) {
            pre = 0;
            traversal(root);
            return root;
        }
    }
}

File: 卡码/代码随想录/1/06-二叉树/C++/04-二叉搜索树属性/LeetCode_0538.cpp
//538. 把二叉搜索树转换为累加树
#include <iostream>
#include "../TreeNode.h"
using namespace std;

class Solution {
	int pre;
public:
	void traversal(TreeNode* root){
		if(!root) return;
		traversal(root->right);
		root->val+=pre;
		pre=root->val;
		traversal(root->left);
	}
  TreeNode* convertBST(TreeNode* root) {
		pre=0;
		traversal(root);
		return root;
  }
};

int main(){
  system("pause");
  return 0;
}
